plugins/registry/handler.go

```diff
@@ -200,6 +209,28 @@ func (rh *registryHandler) relay(ctx context.Context, p registryHandlerParams) {
 		}
 	}
 
+	// reject the push request if the model namespace does not already exist in model-backend
+	ctx = withUserUIDAuth(ctx, p.userUID)
+	if isOrganizationRepository {
+		name := fmt.Sprintf("organizations/%s/models/%s", namespace, contentID)
+		if _, err := rh.modelPublicClient.GetOrganizationModel(ctx, &modelPB.GetOrganizationModelRequest{
+			Name: name,
+			View: modelPB.View_VIEW_BASIC.Enum(),
+		}); err != nil {
+			writeStatusNotFoundError(req, w, "Resource namespace not found")
+			return
+		}
+	} else {
+		name := fmt.Sprintf("users/%s/models/%s", namespace, contentID)
+		if _, err := rh.modelPublicClient.GetUserModel(ctx, &modelPB.GetUserModelRequest{
+			Name: name,
+			View: modelPB.View_VIEW_BASIC.Enum(),
+		}); err != nil {
+			writeStatusNotFoundError(req, w, "Resource namespace not found")
+			return
+		}
+	}
```

Comment:
```suggestion
	if isOrganizationRepository {
		name := fmt.Sprintf("organizations/%s/models/%s", namespace, contentID)
		_, err = rh.modelPublicClient.GetOrganizationModel(ctx, &modelPB.GetOrganizationModelRequest{
			Name: name,
			View: modelPB.View_VIEW_BASIC.Enum(),
		})
	} else {
		name := fmt.Sprintf("users/%s/models/%s", namespace, contentID)
		_, err := rh.modelPublicClient.GetUserModel(ctx, &modelPB.GetUserModelRequest{
			Name: name,
			View: modelPB.View_VIEW_BASIC.Enum(),
		})
	}
	
	if err != nil {
		writeStatusNotFoundError(req, w, "Resource namespace not found")
		return
	}
```

We can group the error handling.